### PR TITLE
chore(dev): make changes to the project to comply with staticcheck

### PR DESCRIPTION
**Pull request by aelath**
## Description

Updating our CLI project to pass staticcheck.

## Motivation and Context

While working in the CLI code I realized there were lots of places that needed some cleanup.

## How has this been tested? How can a reviewer test these changes?

# Have your performed the following tests?

Please confirm you have done the following tests to ensure your pull request is ready to be merged..

- [ ] If this change modifies the deployment flow have triggered a deployment using this branch (i.e. `build/dist/darwin_amd64/armory deploy start -f <path to file>`)?
- [x] If this change modifes the login flow have you tried logging in and out with this branch (i.e. `build/dist/darwin_amd64/armory login`) and verified the credentials work?
- [ ] Have you verified the specific change made in this PR?

### CODE REVIEW

File: pkg/org/org.go [hunk 1]
```diff
@@ -47,43 +42,9 @@ func (e *AppError) String() string {
...
-	var errorResponse *ApiError
-	err = dec.Decode(&errorResponse)
-	if err != nil {
-		return nil, err
-	}
-
-	errContext := fmt.Sprintf(". ErrorId: %s, Desc: %v", errorResponse.ErrorId, errorResponse.Errors)
-	return nil, errorUtils.NewErrorWithDynamicContext(ErrEnvironmentRetrieval, errContext)
-}
-
```
Comment (aelath) [inline pkg/org/org.go:86]: Deleted the environments function as its no longer used. Before this PR it was used by `cmd/login.go`, but I've refactored that code to use the new cloud client in configuration package.

File: cmd/agent/agent_create.go [hunk 2]
```diff
@@ -226,7 +227,7 @@ func (o *AgentOptions) Run() error {
 		}
 
 		if _, err := promptRecreateCredentials.Run(); err != nil {
-			return errors.New(fmt.Sprintf("Exiting %s\n", err))
+			return fmt.Errorf("exiting: %s", err)
```
Comment (Greg-At-Armory) [inline cmd/agent/agent_create.go]: consider using %w for error format

File: cmd/agent/agent_create.go [hunk 3]
```diff
@@ -542,7 +543,7 @@ func (o *AgentOptions) generateManifests() (string, error) {
 		return "", errors.New("unable to parse the manifest template")
```
Comment (Greg-At-Armory) [inline cmd/agent/agent_create.go]: instead of errors.New consider using predefined error variable - just like one from errors.go

Comment (Greg-At-Armory) [reply cmd/agent/agent_create.go]: actually, i see that in a file down below we've got errorUtils.WrappedError, should we unify this?

File: cmd/login/login.go [hunk 4]
```diff
@@ -87,12 +89,18 @@ func login(cmd *cobra.Command, configuration *config.Configuration, envName stri
 		return errorUtils.NewWrappedError(ErrDecodingJwt, err)
 	}
 
-	selectedEnv, err := selectEnvironment(configuration.GetArmoryCloudAddr(), response.AccessToken, envName)
+	err = writeCredentialToFile(cli, parsedJwt, response)
+	if err != nil {
+		return err
+	}
+
+	CloudClient := configuration.NewClient(cli)
```
Comment (Greg-At-Armory) [inline cmd/login/login.go]: any reason why "CloudClient" instead of cloudClient ?

File: cmd/login/login.go [hunk 5]
```diff
@@ -158,7 +168,8 @@ func selectEnvironment(armoryCloudAddr *url.URL, accessToken string, namedEnviro
 		if requestedEnv != nil {
 			return requestedEnv, nil
 		}
-		return nil, errors.New(fmt.Sprintf("Tenant %s not found, please choose a known tenant: [%s]", namedEnvironment[0], strings.Join(environmentNames[:], ",")))
+		//lint:ignore ST1005 errors are user facing
+		return nil, fmt.Errorf("Tenant %s not found, please choose a known tenant: [%s]", namedEnvironment[0], strings.Join(environmentNames[:], ","))
```
Comment (Greg-At-Armory) [inline cmd/login/login.go]: why do we create environmentNames[:] slice?? wouldn't plain environmentNames passed in as arg wouldn't be enough?

File: cmd/login/login.go [hunk 6]
```diff
@@ -170,7 +181,8 @@ func selectEnvironment(armoryCloudAddr *url.URL, accessToken string, namedEnviro
 	_, requestedEnv, err := prompt.Run()
 
 	if err != nil {
-		return nil, errors.New(fmt.Sprintf("failed to select an tenant to login to; %v\n", err))
+		//lint:ignore ST1005 errors are user facing
+		return nil, fmt.Errorf("Failed to select an tenant to login to; %v\n", err)
```
Comment (Greg-At-Armory) [inline cmd/login/login.go]: i recon %w is preffered over %v... i guess :)

File: pkg/org/org.go [hunk 7]
```diff
@@ -93,7 +54,7 @@ func GetAgents(ArmoryCloudAddr *url.URL, accessToken string) ([]Agent, error) {
 	request := util.NewHttpRequest("GET", connectedAgentsUrl.String(), nil, &accessToken)
 	resp, err := request.Execute()
 	if err != nil {
-		return nil, errors.New("Unable to retrieve Remote Network Agents to connect with. Please ensure a Remote Network Agent is connected and try again")
+		return nil, errors.New("unable to retrieve Remote Network Agents to connect with; please ensure a Remote Network Agent is connected and try again")
```
Comment (Greg-At-Armory) [inline pkg/org/org.go]: could you move it to static errors section? like with error utils?

File: pkg/util/httpUtils.go [hunk 8]
```diff
@@ -34,7 +34,9 @@ func NewHttpRequest(method string, url string, body map[string]string, bearerTok
 
 func (request *HttpRequest) Execute() (*http.Response, error) {
 	requestBody, err := json.Marshal(&request.Body)
-
+	if err != nil {
+		return nil, errors.New("failed to marshal request to json")
```
Comment (Greg-At-Armory) [inline pkg/util/httpUtils.go]: as comment above

File: cmd/agent/agent_create.go (hunk 3 above)
Comment (aelath) [reply cmd/agent/agent_create.go]: I added a couple new error vars to the top of the file. That seems to be the pattern in this pkg? 乁(ツ)ㄏ

File: cmd/login/login.go (hunk 4 above)
Comment (aelath) [reply cmd/login/login.go]: Nope, good catch. 👍

File: pkg/org/org.go (hunk 7 above)
Comment (aelath) [reply pkg/org/org.go]: Yep, good call. I made a static error var and moved it to `org/errors.go`

**Review by Greg-At-Armory — APPROVED** [review]

**Comment by aelath** [comment]
I want to test one more thing before I merge this. I'm concerned about some edge cases with the `armory login` functionality.
